File: app/src/main/java/com/teambrella/android/ui/claim/IClaimActivity.java
package com.teambrella.android.ui.claim;

import android.support.annotation.StringRes;

import com.teambrella.android.data.base.IDataHost;

/**
 * Claim activity interface
 */
public interface IClaimActivity extends IDataHost {

    void setTitle(String title);

    void setSubtitle(String subtitle);

    void postVote(int vote);

    String getCurrency();


    void showSnackBar(@StringRes int text);

}

File: app/src/main/java/com/teambrella/android/ui/claim/IClaimActivity.kt
package com.teambrella.android.ui.claim

import android.content.Intent
import android.support.annotation.StringRes

import com.teambrella.android.data.base.IDataHost

/**
 * Claim activity interface
 */
interface IClaimActivity : IDataHost {

    val claimId: Int

    val teamId: Int

    fun setTitle(title: String)

    fun setSubtitle(subtitle: String)

    fun postVote(vote: Int)

    fun showSnackBar(@StringRes text: Int)

    fun launchActivity(intent: Intent)

}
